src/include/gdrwrap.h

```diff
@@ -160,8 +161,18 @@ typedef struct gdr_mem_desc {
 
 #if defined(__HIP_PLATFORM_AMD__) || defined(__HIPCC__)
 static gdr_t ncclGdrInit() {
-  INFO(NCCL_INIT, "Enabled GDRCopy equivalent memory allocation");
-  return (gdr_t)0x12345678L;
+  cudaDeviceProp devProp;
```

Comment:
Since this is guaranteed to be in the HIP block, why not just use HIP? In the same file, there are direct hip usages.

Reply:
In general, if the CUDA API can be hipified, then using CUDA names has the benefit of direct compilation on CUDA.

Reply:
hipFree was missed when we switch to hipify path a few years ago. I will change it as well